pkg/repro/repro.go

```diff
@@ -493,3 +721,24 @@ func (ctx *context) returnInstance(inst *instance) {
 	ctx.bootRequests <- inst.index
 	inst.Close()
 }
+
+func reverseEntries(entries []*prog.LogEntry) []*prog.LogEntry {
+	last := len(entries) - 1
+	for i := 0; i < len(entries)/2; i++ {
+		entries[i], entries[last-i] = entries[last-i], entries[i]
+	}
+	return entries
+}
+
+func encodeEntries(entries []*prog.LogEntry) []byte {
+	var pstr []byte
```

Comment:
I think this will be simpler with `bytes.Buffer`:
```
func encodeEntries(entries []*prog.LogEntry) []byte {
	buf := new(bytes.Buffer)
	for _, ent := range entries {
		opts := ""
		if ent.Fault {
			opts = fmt.Sprintf(" (fault-call:%v fault-nth:%v)", ent.FaultCall, ent.FaultNth)
		}
		fmt.Fprintf(buf, "executing program %v%v:\n%v", ent.Proc, opts, ent.P.Serialize())
	}
	return buf.Bytes()
}
```

Also prefer `range` and `%v` for format strings (continues to work when types change in future).